File: _includes/code/algorithms/cses/python/repetitions.py
from unittest import TestCase
from utils import *


def scanner():
    return get_line()


def algorithm(s):
    if len(s) == 0 || len(s) == 1:
        return 0
    s_shifted = s[1:]
    zipped = zip(s, s_shifted)
    curr = 1
    max_so_far = 1
    for l, r in zipped:
        max_so_far = max(max_so_far, curr)
        if l == r:
            curr += 1
        else:
            curr = 1

    return max_so_far


def test_1():
    assert algorithm('ATTCGGGA') == 3


def test_2():
    assert algorithm('') == 0


def test_3():
    assert algorithm('AT') == 1


def test_4():
    assert algorithm('T') == 1


if __name__ == "__main__":
    print(algorithm(''))
    print(algorithm(scanner()))

File: _includes/code/algorithms/cses/rust/src/problems/introductory/repetitions.rs
use utils::Scanner;

pub fn algorithm(chars: Vec<char>) -> u64 {
    if chars.len() == 0 {
        return 0;
    }

    let mut curr = 1;
    let mut total = 1;
    for (left, right) in chars.iter().zip(chars.iter().skip(1)) {
        curr = if left == right { curr + 1 } else { 1 };
        total = std::cmp::max(curr, total);
    }
    total
}

#[allow(dead_code)]
pub fn answer() {
    let stdin = std::io::stdin();
    let mut scanner = Scanner::new(stdin.lock());

    let sequence = scanner.split_line();

    let total = algorithm(sequence);

    println!("{:?}", total);
}

#[test]
fn test_1() {
    assert_eq!(algorithm("ATTCGGGA".chars().collect()), 3);
}
#[test]
fn test_2() {
    assert_eq!(algorithm("ATTTTCGGGA".chars().collect()), 4);
}

#[test]
fn test_3() {
    assert_eq!(algorithm("ATTCGGA".chars().collect()), 2);
}
